Cached prompt not updated immediately for git worktree

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### What happened?

Though I'm positive I tested this (perhaps not?), my changes to effectively invalidate the cache for the `git` segment don't work for worktrees. That is, if I switch branches, the cache timeout still applies. This works correctly for primary repos.

The problem is in https://github.com/JanDeDobbeleer/oh-my-posh/blob/dd9fbb6dd95c919b2adaa248c6d3ce750eb4885e/src/segments/git.go#L227. I expected `dir.Path` at that point to be the resolved primary repo or worktree thereunder directory, but isn't. I did a lot of refactoring to reduce duplicate code and reduce file reads, so at least a few iterations I know it worked but maybe the last iteration I committed has the ordering wrong.

That is, a worktree's `.git` root *file* will point to the primary repo's `.git/worktrees/{worktree-name}/` directory and that's where we need to read the `HEAD` file.

### Theme

<details><summary>theme.omp.yml</summary>
<p>

```yaml
# yaml-language-server: $schema=https://raw.githubusercontent.com/JanDeDobbeleer/oh-my-posh/main/themes/schema.json
version: 3
console_title_template: >-
  {{ $pwd := .PWD -}}
  {{ .Shell }} in {{ with $repo := .Segments.Git -}}
    {{ if (gt (sub (len $pwd) (len $repo.Dir)) 20) -}}
      git:{{ $repo.RepoName }}/ΓÇª/{{ base $pwd }}
    {{- else -}}
      git:{{ $repo.RepoName }}{{ trimPrefix $repo.Dir $pwd}}
    {{- end }}
  {{- else -}}
    {{ .Folder }}
  {{- end }}
final_space: true
shell_integration: true

palette:
  # https://colorkit.co/palette/02944e-4f0193-f1c516-eb850d-e44403-05a3e2-f7f6f5-7c7b7b-464646-101010/
  green: "#02944e"
  purple: "#4f0193"
  darkPurple: "#940248" # not in palette
  yellow: "#f1c516"
  orange: "#eb850d"
  red: "#e44403"
  blue: "#05a3e2"
  white: "#f7f6f5"
  gray: "#7c7b7b"
  darkGray: "#464646"
  black: "#101010"

blocks:
- type: prompt
  alignment: left
  segments:
  - type: shell
    style: diamond
    foreground: p:white
    background: p:purple
    properties:
      mapped_shell_names:
        powershell: PS
        pwsh: PS

  - type: python
    style: powerline
    powerline_symbol: "εé░"
    foreground: p:white
    background: p:darkPurple
    template: " εê╡ {{ if .Error }}{{ .Error }}{{ else }}{{ if .Venv }}{{ .Venv }} {{ end }}{{ .Full }}{{ end }} "

  - type: executiontime
    style: powerline
    powerline_symbol: "εé░"
    foreground: p:black
    background: p:orange
    properties:
      style: austin

  - type: status
    style: powerline
    powerline_symbol: "εé░"
    foreground: p:white
    background: p:blue
    background_templates:
    - "{{ if gt .Code 0 }}p:red{{ end }}"
    properties:
      always_enable: true

  - type: session
    style: powerline
    powerline_symbol: "εé░"
    foreground: p:white
    background: p:blue
    template: " {{ if .SSHSession }}∩áù {{ end }}{{ .UserName }} "

  - type: root
    style: powerline
    powerline_symbol: "εé░"
    foreground: p:white
    background: p:red
    template: " ε¡┤ "

  - type: path
    style: powerline
    powerline_symbol: "εé░"
    foreground: p:black
    background: p:gray
    template: ' {{ path .Path .Location }} {{ repeat .StackCount "εé▒" }}'
    properties:
      style: full

- type: prompt
  alignment: left
  newline: true
  segments:
  - type: git
    cache:
      duration: 30s
      strategy: folder
    style: diamond
    trailing_diamond: "εé░"
    foreground: p:white
    foreground_templates:
    - "{{ if or (.Working.Changed) (.Staging.Changed) }}p:black{{ end }}"
    - "{{ if and (gt .Ahead 0) (gt .Behind 0) }}p:white{{ end }}"
    - "{{ if gt .Ahead 0 }}p:black{{ end }}"
    background: p:green
    background_templates:
    - "{{ if or (.Working.Changed) (.Staging.Changed) }}p:yellow{{ end }}"
    - "{{ if and (gt .Ahead 0) (gt .Behind 0) }}p:red{{ end }}"
    - "{{ if gt .Ahead 0 }}p:yellow{{ end }}"
    - "{{ if gt .Behind 0 }}p:orange{{ end }}"
    template: " {{ if .UpstreamURL }}{{ url .UpstreamIcon .UpstreamURL }}{{ end }}{{
      .HEAD }}{{if .BranchStatus }} {{ .BranchStatus }}{{ end }}{{ if .Working.Changed
      }} ∩üä {{ .Working.String }}{{ end }}{{ if .Staging.Changed }} ∩üå {{ .Staging.String
      }}{{ end }} "
    properties:
      branch_icon: "εéá "
      fetch_status: true

debug_prompt:
  background: transparent
  template: "<p:white,p:red> ε¬» DBG </><p:red,transparent>εé░</> "

secondary_prompt:
  background: transparent
  template: "<p:black,p:gray> εé▒</><p:gray,transparent>εé░</> "

# vim: set et sw=2:
```

</p>
</details> 

### What OS are you seeing the problem on?

Windows, Linux, macOS

### Which shell are you using?

bash

### Log output

```Shell
Version: 24.18.0

Shell: bash (5.1.16(1)-release)

Prompt:

]133;D;0]133;A bash εé░ heaths εé░ ]8;;file:\~/src/autorest.rust/packages/typespec-rust]8;;\ εé░
 εéá main Γëí εé░ ]133;B

Segments:

ConsoleTitle(true)                         -   0 ms
Shell(true)                                -   0 ms
Python(false)                              -  99 ms
Executiontime(false)                       -   0 ms
Status(false)                              -   0 ms
Session(true)                              -   2 ms
Root(false)                                -   0 ms
Path(true)                                 -   1 ms
Git(true)                                  -   0 ms

Run duration: 103.115121ms

Cache path: /home/heaths/.cache/oh-my-posh

Config path: /home/heaths/.config/oh-my-posh/theme.omp.yml

Logs:

[38;2;191;207;240m[DEBUG] [38;2;156;231;201m17:07:29.847 [0mcommand.go:execute:989[38;2;156;231;201m ΓåÆ[0m debug mode enabled
[38;2;204;137;214m[TRACE] [38;2;156;231;201m17:07:29.847 [0mdebug.go() - [38;2;156;231;201m1.144┬╡s[0m
[38;2;204;137;214m[TRACE] [38;2;156;231;201m17:07:29.848 [0mload.go:Load() - [38;2;156;231;201m1.032287ms[0m
[38;2;204;137;214m[TRACE] [38;2;156;231;201m17:07:29.848 [0mdebug.go() - [38;2;156;231;201m1.045213ms[0m
[DEBUG] 17:07:29.848 debug.go:52 ΓåÆ plain mode enabled
[DEBUG] 17:07:29.848 home.go:Home:16 ΓåÆ /home/heaths
[TRACE] 17:07:29.848 terminal.go() - 12.847┬╡s
[DEBUG] 17:07:29.848 terminal.go:59 ΓåÆ loading cache file: /home/heaths/.cache/oh-my-posh/omp.cache
[DEBUG] 17:07:29.848 terminal.go:59 ΓåÆ loading cache key: is_wsl
[DEBUG] 17:07:29.848 terminal.go:59 ΓåÆ loading cache key: environment_platform
[TRACE] 17:07:29.848 terminal.go(/home/heaths/.cache/oh-my-posh/omp.cache) - 51.511┬╡s
[DEBUG] 17:07:29.848 home.go:Home:16 ΓåÆ /home/heaths
[TRACE] 17:07:29.848 terminal.go() - 6.068┬╡s
[DEBUG] 17:07:29.848 terminal.go:59 ΓåÆ loading cache file: /home/heaths/.cache/oh-my-posh/omp.cache.a2ca2487-dcf9-481e-a5d9-52d1d14c703d
[DEBUG] 17:07:29.848 terminal.go:59 ΓåÆ loading cache key: prompt_count_cache
[DEBUG] 17:07:29.848 terminal.go:59 ΓåÆ loading cache key: segment_cache_Git_/home/heaths/src/autorest.rust/.git@main
[DEBUG] 17:07:29.848 terminal.go:59 ΓåÆ loading cache key: template_cache
[TRACE] 17:07:29.848 terminal.go(/home/heaths/.cache/oh-my-posh/omp.cache.a2ca2487-dcf9-481e-a5d9-52d1d14c703d) - 80.278┬╡s
[TRACE] 17:07:29.848 terminal.go:Init() - 347ns
[TRACE] 17:07:29.848 terminal.go() - 37ns
[DEBUG] 17:07:29.848 terminal.go:Init:67 ΓåÆ /home/heaths/src/autorest.rust/packages/typespec-rust
[TRACE] 17:07:29.848 terminal.go:Init() - 10.362┬╡s
[TRACE] 17:07:29.848 debug.go() - 203.745┬╡s
[TRACE] 17:07:29.848 init.go:Init() - 53ns
[TRACE] 17:07:29.848 cache.go:loadCache() - 37ns
[TRACE] 17:07:29.848 cache.go:loadCache() - 84.564┬╡s
[DEBUG] 17:07:29.848 debug.go:61 ΓåÆ terminal program: Windows Terminal
[DEBUG] 17:07:29.848 debug.go:61 ΓåÆ terminal shell: shell
[DEBUG] 17:07:29.848 config.go:MakeColors:58 ΓåÆ NO DATA
[TRACE] 17:07:29.848 config.go:MakeColors(OMP_CACHE_DISABLED) - 5.512┬╡s
[TRACE] 17:07:29.849 colors.go:MakeColors() - 643ns
[TRACE] 17:07:29.849 debug.go:PrintDebug() - 96ns
[DEBUG] 17:07:29.849 debug.go:PrintDebug:16 ΓåÆ 5.1.16(1)-release
[TRACE] 17:07:29.849 debug.go:PrintDebug(POSH_SHELL_VERSION) - 8.421┬╡s
[DEBUG] 17:07:29.849 debug.go:72 ΓåÆ segment: Title
[TRACE] 17:07:29.849 engine.go:getTitleTemplateText({{ $pwd := .PWD -}} {{ .Shell }} in {{ with $repo := .Segments.Git -}}
  {{ if (gt (sub (len $pwd) (len $repo.Dir)) 20) -}}
    git:{{ $repo.RepoName }}/ΓÇª/{{ base $pwd }}
  {{- else -}}
    git:{{ $repo.RepoName }}{{ trimPrefix $repo.Dir $pwd}}
  {{- end }}
{{- else -}}
  {{ .Folder }}
{{- end }}) - 342.77┬╡s
[TRACE] 17:07:29.849 primary.go:needsPrimaryRightPrompt() - 43ns
[TRACE] 17:07:29.849 primary.go:writePrimaryPrompt() - 130ns
[TRACE] 17:07:29.849 primary.go:writePrimaryPrompt() - 23ns
[DEBUG] 17:07:29.849 terminal.go:CursorPosition:583 ΓåÆ NO DATA
[TRACE] 17:07:29.849 terminal.go:CursorPosition(POSH_CURSOR_LINE) - 6.171┬╡s
[DEBUG] 17:07:29.849 terminal.go:CursorPosition:587 ΓåÆ NO DATA
[TRACE] 17:07:29.849 terminal.go:CursorPosition(POSH_CURSOR_COLUMN) - 2.511┬╡s
[TRACE] 17:07:29.849 primary.go:writePrimaryPrompt() - 34ns
[TRACE] 17:07:29.849 primary.go:writePrimaryPrompt() - 23ns
[TRACE] 17:07:29.849 segment.go:Execute() - 238ns
[DEBUG] 17:07:29.849 home.go:Home:16 ΓåÆ /home/heaths
[TRACE] 17:07:29.849 segment.go:Execute() - 209ns
[TRACE] 17:07:29.849 terminal.go:DirMatchesOneOf() - 79ns
[DEBUG] 17:07:29.849 home.go:Home:16 ΓåÆ /home/heaths
[DEBUG] 17:07:29.849 segment.go:Execute:110 ΓåÆ segment: Path
[TRACE] 17:07:29.849 terminal.go:DirMatchesOneOf() - 76ns
[DEBUG] 17:07:29.849 segment.go:Execute:112 ΓåÆ no toggles found
[DEBUG] 17:07:29.849 segment.go:Execute:110 ΓåÆ segment: Executiontime
[DEBUG] 17:07:29.849 segment.go:Execute:112 ΓåÆ no toggles found
[TRACE] 17:07:29.849 segment.go:Execute() - 303ns
[DEBUG] 17:07:29.849 home.go:Home:16 ΓåÆ /home/heaths
[TRACE] 17:07:29.849 terminal.go:DirMatchesOneOf() - 78ns
[DEBUG] 17:07:29.849 segment.go:Execute:110 ΓåÆ segment: Shell
[DEBUG] 17:07:29.849 segment.go:Execute:112 ΓåÆ no toggles found
[DEBUG] 17:07:29.849 executiontime.go:Enabled:55 ΓåÆ always_enabled: false
[DEBUG] 17:07:29.849 shell.go:Enabled:26 ΓåÆ mapped_shell_names: map[powershell:PS pwsh:PS]
[TRACE] 17:07:29.849 executiontime.go:Enabled() - 124ns
[TRACE] 17:07:29.849 shell.go:Enabled() - 87ns
[TRACE] 17:07:29.849 shell.go:Enabled() - 51ns
[DEBUG] 17:07:29.849 executiontime.go:Enabled:57 ΓåÆ threshold: 500.000000
[TRACE] 17:07:29.849 segment.go:Execute() - 42ns
[TRACE] 17:07:29.849 segment.go:Execute() - 96ns
[DEBUG] 17:07:29.849 path.go:147 ΓåÆ display_cygpath: false
[DEBUG] 17:07:29.849 home.go:Home:16 ΓåÆ /home/heaths
[TRACE] 17:07:29.849 path.go:setPaths() - 122ns
[DEBUG] 17:07:29.849 home.go:Home:16 ΓåÆ /home/heaths
[TRACE] 17:07:29.849 terminal.go:DirMatchesOneOf() - 48ns
[TRACE] 17:07:29.849 path.go:setPaths() - 76ns
[DEBUG] 17:07:29.849 segment.go:Execute:110 ΓåÆ segment: Status
[DEBUG] 17:07:29.849 segment.go:Execute:110 ΓåÆ segment: Python
[TRACE] 17:07:29.849 path.go:setPaths() - 126ns
[TRACE] 17:07:29.849 status.go:Enabled() - 126ns
[TRACE] 17:07:29.849 path.go:setPaths() - 40ns
[DEBUG] 17:07:29.849 status.go:formatStatus:45 ΓåÆ {{ .Code }}
[DEBUG] 17:07:29.849 python.go:Enabled:57 ΓåÆ environment
[TRACE] 17:07:29.849 path.go:parsePath() - 39ns
[DEBUG] 17:07:29.849 path.go:setMappedLocations:557 ΓåÆ mapped_locations_enabled: true
[DEBUG] 17:07:29.849 path.go:setMappedLocations:558 ΓåÆ ∩Çô
[DEBUG] 17:07:29.849 language.go:Enabled:118 ΓåÆ extensions: [*.py *.ipynb pyproject.toml venv.bak]
[DEBUG] 17:07:29.849 path.go:setMappedLocations:559 ΓåÆ ∩Çô
[DEBUG] 17:07:29.849 language.go:Enabled:119 ΓåÆ folders: [.venv venv virtualenv venv-win pyenv-win]
[DEBUG] 17:07:29.849 language.go:Enabled:126 ΓåÆ home_enabled: false
[DEBUG] 17:07:29.849 home.go:Home:16 ΓåÆ /home/heaths
[DEBUG] 17:07:29.849 python.go:loadContext:65 ΓåÆ fetch_virtual_env: true
[DEBUG] 17:07:29.849 home.go:Home:16 ΓåÆ /home/heaths
[TRACE] 17:07:29.849 clean.go:Clean() - 134ns
[TRACE] 17:07:29.849 path.go:normalize() - 58ns
[TRACE] 17:07:29.849 path.go:normalize() - 29ns
[DEBUG] 17:07:29.849 path.go:setMappedLocations:560 ΓåÆ ~
[DEBUG] 17:07:29.849 path.go:setMappedLocations:565 ΓåÆ mapped_locations: map[]
[TRACE] 17:07:29.849 clean.go:Clean() - 56ns
[TRACE] 17:07:29.849 path.go:normalize() - 35ns
[TRACE] 17:07:29.849 path.go:normalize() - 39ns
[TRACE] 17:07:29.849 clean.go:Clean() - 38ns
[TRACE] 17:07:29.849 path.go:normalize() - 40ns
[TRACE] 17:07:29.849 path.go:normalize() - 34ns
[TRACE] 17:07:29.849 path.go:parsePath() - 38ns
[TRACE] 17:07:29.849 path.go:parsePath() - 34ns
[TRACE] 17:07:29.849 path.go:parsePath() - 32ns
[DEBUG] 17:07:29.849 path.go:makeFolderFormatMap:833 ΓåÆ NO DATA
[DEBUG] 17:07:29.850 path.go:setStyle:213 ΓåÆ full
[DEBUG] 17:07:29.850 path.go:colorizePath:734 ΓåÆ cycle: []
[DEBUG] 17:07:29.850 path.go:getFolderSeparator:267 ΓåÆ NO DATA
[DEBUG] 17:07:29.850 path.go:getFolderSeparator:269 ΓåÆ /
[TRACE] 17:07:29.850 segment.go:Execute() - 198ns
[TRACE] 17:07:29.850 status.go:formatStatus({{ .Code }}) - 330.887┬╡s
[DEBUG] 17:07:29.850 status.go:Enabled:37 ΓåÆ always_enabled: false
[DEBUG] 17:07:29.850 path.go:colorizePath:737 ΓåÆ cycle_folder_separator: false
[TRACE] 17:07:29.850 segment.go:Execute() - 84ns
[DEBUG] 17:07:29.850 home.go:Home:16 ΓåÆ /home/heaths
[TRACE] 17:07:29.850 terminal.go:DirMatchesOneOf() - 212ns
[DEBUG] 17:07:29.850 home.go:Home:16 ΓåÆ /home/heaths
t: Session
[DEBUG] 17:07:29.850 segment.go:Execute:112 ΓåÆ no toggles found
[TRACE] 17:07:29.850 terminal.go:DirMatchesOneOf() - 135ns
[DEBUG] 17:07:29.850 session.go:activeSSHSession:31 ΓåÆ NO DATA
[DEBUG] 17:07:29.850 segment.go:Execute:110 ΓåÆ segment: Root
[TRACE] 17:07:29.850 session.go:activeSSHSession(SSH_CONNECTION) - 4.963┬╡s
[DEBUG] 17:07:29.850 segment.go:Execute:112 ΓåÆ no toggles found
[DEBUG] 17:07:29.850 session.go:activeSSHSession:31 ΓåÆ NO DATA
[TRACE] 17:07:29.850 session.go:activeSSHSession(SSH_CLIENT) - 2.954┬╡s
[DEBUG] 17:07:29.850 session.go:activeSSHSession:37 ΓåÆ ubuntu
[TRACE] 17:07:29.850 root.go:Enabled() - 502ns
[DEBUG] 17:07:29.850 path.go:colorizePath:738 ΓåÆ %s
[DEBUG] 17:07:29.851 path.go:colorizePath:740 ΓåÆ %s
[DEBUG] 17:07:29.851 path.go:colorizePath:741 ΓåÆ %s
[DEBUG] 17:07:29.851 path.go:colorizePath:742 ΓåÆ %s
[TRACE] 17:07:29.851 segment.go:string( {{ .Name }} ) - 1.168253ms
[TRACE] 17:07:29.851 path.go:Enabled() - 112ns
[TRACE] 17:07:29.851 path.go:Enabled() - 67ns
[TRACE] 17:07:29.851 segment.go:resolve(diamond) - 269ns
[TRACE] 17:07:29.851 path.go:Enabled() - 56ns
[TRACE] 17:07:29.851 path.go:Enabled(/home/heaths/src/autorest.rust/packages/typespec-rust) - 32.582┬╡s
[DEBUG] 17:07:29.852 session.go:activeSSHSession:41 ΓåÆ NO DATA
[TRACE] 17:07:29.852 session.go:activeSSHSession(who am i) - 1.147924ms
[ERROR] 17:07:29.947 python.go:pyvenvCfgPrompt:179 ΓåÆ exec: "python": executable file not found in $PATH
[TRACE] 17:07:29.947 python.go:pyvenvCfgPrompt(python) - 97.890646ms
[DEBUG] 17:07:29.947 python.go:pyvenvCfgPrompt:181 ΓåÆ /usr/bin/python3
[TRACE] 17:07:29.947 python.go:pyvenvCfgPrompt(python3) - 76.788┬╡s
[DEBUG] 17:07:29.949 python.go:pyvenvCfgPrompt:189 ΓåÆ false
[TRACE] 17:07:29.949 python.go:pyvenvCfgPrompt(pyvenv.cfg) - 1.440055ms
[DEBUG] 17:07:29.949 python.go:pyvenvCfgPrompt:193 ΓåÆ false
[TRACE] 17:07:29.949 python.go:pyvenvCfgPrompt(pyvenv.cfg) - 33.301┬╡s
[DEBUG] 17:07:29.949 python.go:loadContext:79 ΓåÆ folder_name_fallback: true
[DEBUG] 17:07:29.949 python.go:loadContext:80 ΓåÆ default_venv_names: [.venv venv]
[DEBUG] 17:07:29.949 python.go:loadContext:87 ΓåÆ NO DATA
[TRACE] 17:07:29.949 python.go:loadContext(VIRTUAL_ENV) - 4.936┬╡s
[DEBUG] 17:07:29.949 python.go:loadContext:87 ΓåÆ NO DATA
[TRACE] 17:07:29.949 python.go:loadContext(CONDA_ENV_PATH) - 3.299┬╡s
[DEBUG] 17:07:29.949 python.go:loadContext:87 ΓåÆ NO DATA
[TRACE] 17:07:29.949 python.go:loadContext(CONDA_DEFAULT_ENV) - 2.658┬╡s
[TRACE] 17:07:29.949 segment.go:resolve(powerline) - 813ns
[TRACE] 17:07:29.949 segment.go:resolve(powerline) - 153ns
[TRACE] 17:07:29.949 segment.go:resolve(powerline) - 121ns
[TRACE] 17:07:29.949 segment.go:string( {{ if .SSHSession }}∩áù {{ end }}{{ .UserName }} ) - 78.488┬╡s
[TRACE] 17:07:29.949 segment.go:resolve(powerline) - 206ns
[TRACE] 17:07:29.949 segment.go:resolve(powerline) - 123ns
[TRACE] 17:07:29.949 segment.go:string( {{ path .Path .Location }} {{ repeat .StackCount "εé▒" }}) - 58.826┬╡s
[TRACE] 17:07:29.949 segment.go:resolve(powerline) - 210ns
[TRACE] 17:07:29.949 engine.go:applyPowerShellBleedPatch() - 307ns
[TRACE] 17:07:29.949 engine.go:applyPowerShellBleedPatch() - 37ns
[TRACE] 17:07:29.949 segment.go:Execute() - 84ns
[DEBUG] 17:07:29.949 home.go:Home:16 ΓåÆ /home/heaths
[TRACE] 17:07:29.949 terminal.go:DirMatchesOneOf() - 54ns
[DEBUG] 17:07:29.949 segment.go:Execute:110 ΓåÆ segment: Git
[DEBUG] 17:07:29.949 segment.go:Execute:112 ΓåÆ no toggles found
[DEBUG] 17:07:29.949 terminal.go:HasParentFilePath:446 ΓåÆ /home/heaths/src/autorest.rust/packages/typespec-rust
[TRACE] 17:07:29.949 terminal.go:HasParentFilePath(/home/heaths/src/autorest.rust/packages/typespec-rust) - 21.664┬╡s
[TRACE] 17:07:29.949 git.go:CacheKey(.git) - 36.069┬╡s
[DEBUG] 17:07:29.949 scm.go:FileContents:142 Γåô
    ref: refs/heads/main
    
[TRACE] 17:07:29.949 scm.go:FileContents(/home/heaths/src/autorest.rust/.git/HEAD) - 22.519┬╡s
[DEBUG] 17:07:29.949 segment.go:Execute:116 ΓåÆ restored segment from cache:  Git
[TRACE] 17:07:29.950 segment.go:string( {{ if .UpstreamURL }}{{ url .UpstreamIcon .UpstreamURL }}{{ end }}{{ .HEAD }}{{if .BranchStatus }} {{ .BranchStatus }}{{ end }}{{ if .Working.Changed }} ∩üä {{ .Working.String }}{{ end }}{{ if .Staging.Changed }} ∩üå {{ .Staging.String }}{{ end }} ) - 356.532┬╡s
[TRACE] 17:07:29.950 list.go:FirstMatch({{ if or (.Working.Changed) (.Staging.Changed) }}p:yellow{{ end }}) - 36.123┬╡s
[TRACE] 17:07:29.950 list.go:FirstMatch({{ if and (gt .Ahead 0) (gt .Behind 0) }}p:red{{ end }}) - 45.304┬╡s
[TRACE] 17:07:29.950 list.go:FirstMatch({{ if gt .Ahead 0 }}p:yellow{{ end }}) - 22.783┬╡s
[TRACE] 17:07:29.950 list.go:FirstMatch({{ if gt .Behind 0 }}p:orange{{ end }}) - 23.512┬╡s
[TRACE] 17:07:29.950 list.go:FirstMatch({{ if or (.Working.Changed) (.Staging.Changed) }}p:black{{ end }}) - 35.278┬╡s
[TRACE] 17:07:29.950 list.go:FirstMatch({{ if and (gt .Ahead 0) (gt .Behind 0) }}p:white{{ end }}) - 19.016┬╡s
[TRACE] 17:07:29.950 list.go:FirstMatch({{ if gt .Ahead 0 }}p:black{{ end }}) - 12.773┬╡s
[TRACE] 17:07:29.950 segment.go:resolve(diamond) - 147ns
[TRACE] 17:07:29.950 list.go:FirstMatch({{ if or (.Working.Changed) (.Staging.Changed) }}p:yellow{{ end }}) - 19.697┬╡s
[TRACE] 17:07:29.950 list.go:FirstMatch({{ if and (gt .Ahead 0) (gt .Behind 0) }}p:red{{ end }}) - 16.943┬╡s
[TRACE] 17:07:29.950 list.go:FirstMatch({{ if gt .Ahead 0 }}p:yellow{{ end }}) - 16.943┬╡s
[TRACE] 17:07:29.950 list.go:FirstMatch({{ if gt .Behind 0 }}p:orange{{ end }}) - 12.408┬╡s
[TRACE] 17:07:29.950 list.go:FirstMatch({{ if or (.Working.Changed) (.Staging.Changed) }}p:black{{ end }}) - 27.58┬╡s
[TRACE] 17:07:29.950 list.go:FirstMatch({{ if and (gt .Ahead 0) (gt .Behind 0) }}p:white{{ end }}) - 18.568┬╡s
[TRACE] 17:07:29.950 list.go:FirstMatch({{ if gt .Ahead 0 }}p:black{{ end }}) - 11.932┬╡s
[TRACE] 17:07:29.950 engine.go:applyPowerShellBleedPatch() - 102ns
[TRACE] 17:07:29.950 engine.go:applyPowerShellBleedPatch() - 30ns
[TRACE] 17:07:29.950 primary.go:writePrimaryPrompt() - 43ns
[TRACE] 17:07:29.950 primary.go:Primary() - 26ns
[DEBUG] 17:07:29.950 home.go:Home:16 ΓåÆ /home/heaths
[TRACE] 17:07:29.950 debug.go:PrintDebug() - 15.656┬╡s
[TRACE] 17:07:29.950 debug.go:PrintDebug() - 33ns
```

--- a/src/segments/git.go
+++ b/src/segments/git.go
@@ -224,8 +224,15 @@ func (g *Git) CacheKey() (string, bool) {
 		return "", false
 	}
 
-	ref := g.FileContents(dir.Path, "HEAD")
+	if !g.isRepo(dir) {
+		return "", false
+	}
+
+	ref := g.FileContents(g.mainSCMDir, "HEAD")
 	ref = strings.Replace(ref, "ref: refs/heads/", "", 1)
+
+	// Use the repo clone in the cache key so the mapped path is consistent
+	// for primary and worktree repos.
 	return fmt.Sprintf("%s@%s", dir.Path, ref), true
 }
 
@@ -295,7 +302,7 @@ func (g *Git) StashCount() int {
 		return g.stashCount
 	}
 
-	stashContent := g.FileContents(g.rootDir, "logs/refs/stash")
+	stashContent := g.FileContents(g.scmDir, "logs/refs/stash")
 	if stashContent == "" {
 		return 0
 	}
@@ -335,11 +342,11 @@ func (g *Git) shouldDisplay() bool {
 	}
 
 	if g.props.GetBool(FetchBareInfo, false) {
-		g.realDir = g.env.Pwd()
+		g.repoRootDir = g.env.Pwd()
 		bare := g.getGitCommandOutput("rev-parse", "--is-bare-repository")
 		if bare == trueStr {
 			g.IsBare = true
-			g.workingDir = g.realDir
+			g.mainSCMDir = g.repoRootDir
 			return true
 		}
 	}
@@ -349,21 +356,25 @@ func (g *Git) shouldDisplay() bool {
 		return false
 	}
 
+	return g.isRepo(gitdir)
+}
+
+func (g *Git) isRepo(gitdir *runtime.FileInfo) bool {
 	g.setDir(gitdir.Path)
 
 	if !gitdir.IsDir {
 		if g.hasWorktree(gitdir) {
-			g.realDir = g.convertToWindowsPath(g.realDir)
+			g.repoRootDir = g.convertToWindowsPath(g.repoRootDir)
 			return true
 		}
 
 		return false
 	}
 
-	g.workingDir = gitdir.Path
-	g.rootDir = gitdir.Path
+	g.mainSCMDir = gitdir.Path
+	g.scmDir = gitdir.Path
 	// convert the worktree file path to a windows one when in a WSL shared folder
-	g.realDir = strings.TrimSuffix(g.convertToWindowsPath(gitdir.Path), "/.git")
+	g.repoRootDir = strings.TrimSuffix(g.convertToWindowsPath(gitdir.Path), "/.git")
 	return true
 }
 
@@ -377,10 +388,10 @@ func (g *Git) getBareRepoInfo() {
 	if file, err := g.env.HasParentFilePath(".git", true); err == nil && !file.IsDir {
 		content := g.FileContents(file.ParentFolder, ".git")
 		dir := strings.TrimPrefix(content, "gitdir: ")
-		g.workingDir = filepath.Join(file.ParentFolder, dir)
+		g.mainSCMDir = filepath.Join(file.ParentFolder, dir)
 	}
 
-	head := g.FileContents(g.workingDir, "HEAD")
+	head := g.FileContents(g.mainSCMDir, "HEAD")
 	branchIcon := g.props.GetString(BranchIcon, "\uE0A0")
 	g.Ref = strings.Replace(head, "ref: refs/heads/", "", 1)
 	g.HEAD = fmt.Sprintf("%s%s", branchIcon, g.formatBranch(g.Ref))
@@ -405,7 +416,7 @@ func (g *Git) setDir(dir string) {
 }
 
 func (g *Git) hasWorktree(gitdir *runtime.FileInfo) bool {
-	g.rootDir = gitdir.Path
+	g.scmDir = gitdir.Path
 	content := g.env.FileContent(gitdir.Path)
 	content = strings.Trim(content, " \r\n")
 	matches := regex.FindNamedRegexMatch(`^gitdir: (?P<dir>.*)$`, content)
@@ -417,20 +428,20 @@ func (g *Git) hasWorktree(gitdir *runtime.FileInfo) bool {
 
 	// if we open a worktree file in a WSL shared folder, we have to convert it back
 	// to the mounted path
-	g.workingDir = g.convertToLinuxPath(matches["dir"])
+	g.mainSCMDir = g.convertToLinuxPath(matches["dir"])
 
 	// if we don't do this, we will identify the submodule as a worktree
-	isSubmodule := strings.Contains(g.workingDir, "/modules/")
+	isSubmodule := strings.Contains(g.mainSCMDir, "/modules/")
 
 	// in worktrees, the path looks like this: gitdir: path/.git/worktrees/branch
-	// rootDir needs to become path/.git
-	// realDir needs to become path
-	ind := strings.LastIndex(g.workingDir, "/worktrees/")
+	// scmDir needs to become path/.git
+	// repoRootDir needs to become path
+	ind := strings.LastIndex(g.mainSCMDir, "/worktrees/")
 	if ind > -1 && !isSubmodule {
-		gitDir := filepath.Join(g.workingDir, "gitdir")
-		g.rootDir = g.workingDir[:ind]
+		gitDir := filepath.Join(g.mainSCMDir, "gitdir")
+		g.scmDir = g.mainSCMDir[:ind]
 		gitDirContent := g.env.FileContent(gitDir)
-		g.realDir = strings.TrimSuffix(gitDirContent, ".git\n")
+		g.repoRootDir = strings.TrimSuffix(gitDirContent, ".git\n")
 		g.IsWorkTree = true
 		return true
 	}
@@ -438,33 +449,33 @@ func (g *Git) hasWorktree(gitdir *runtime.FileInfo) bool {
 	// in submodules, the path looks like this: gitdir: ../.git/modules/test-submodule
 	// we need the parent folder to detect where the real .git folder is
 	if isSubmodule {
-		g.rootDir = resolveGitPath(gitdir.ParentFolder, g.workingDir)
+		g.scmDir = resolveGitPath(gitdir.ParentFolder, g.mainSCMDir)
 		// this might be both a worktree and a submodule, where the path would look like
 		// this: path/.git/modules/module/path/worktrees/location. We cannot distinguish
 		// between worktree and a module path containing the word 'worktree,' however.
-		ind = strings.LastIndex(g.rootDir, "/worktrees/")
-		if ind > -1 && g.env.HasFilesInDir(g.rootDir, "gitdir") {
-			gitDir := filepath.Join(g.rootDir, "gitdir")
+		ind = strings.LastIndex(g.scmDir, "/worktrees/")
+		if ind > -1 && g.env.HasFilesInDir(g.scmDir, "gitdir") {
+			gitDir := filepath.Join(g.scmDir, "gitdir")
 			realGitFolder := g.env.FileContent(gitDir)
-			g.realDir = strings.TrimSuffix(realGitFolder, ".git\n")
-			g.rootDir = g.rootDir[:ind]
-			g.workingDir = g.rootDir
+			g.repoRootDir = strings.TrimSuffix(realGitFolder, ".git\n")
+			g.scmDir = g.scmDir[:ind]
+			g.mainSCMDir = g.scmDir
 			g.IsWorkTree = true
 			return true
 		}
 
-		g.realDir = g.rootDir
-		g.workingDir = g.rootDir
+		g.repoRootDir = g.scmDir
+		g.mainSCMDir = g.scmDir
 		return true
 	}
 
 	// check for separate git folder(--separate-git-dir)
 	// check if the folder contains a HEAD file
-	if g.env.HasFilesInDir(g.workingDir, "HEAD") {
-		gitFolder := strings.TrimSuffix(g.rootDir, ".git")
-		g.rootDir = g.workingDir
-		g.workingDir = gitFolder
-		g.realDir = gitFolder
+	if g.env.HasFilesInDir(g.mainSCMDir, "HEAD") {
+		gitFolder := strings.TrimSuffix(g.scmDir, ".git")
+		g.scmDir = g.mainSCMDir
+		g.mainSCMDir = gitFolder
+		g.repoRootDir = gitFolder
 		return true
 	}
 
@@ -473,7 +484,7 @@ func (g *Git) hasWorktree(gitdir *runtime.FileInfo) bool {
 
 func (g *Git) shouldIgnoreStatus() bool {
 	list := g.props.GetStringArray(IgnoreStatus, []string{})
-	return g.env.DirMatchesOneOf(g.realDir, list)
+	return g.env.DirMatchesOneOf(g.repoRootDir, list)
 }
 
 func (g *Git) setBranchStatus() {
@@ -668,7 +679,7 @@ func (g *Git) setGitStatus() {
 }
 
 func (g *Git) getGitCommandOutput(args ...string) string {
-	args = append([]string{"-C", g.realDir, "--no-optional-locks", "-c", "core.quotepath=false", "-c", "color.status=false"}, args...)
+	args = append([]string{"-C", g.repoRootDir, "--no-optional-locks", "-c", "core.quotepath=false", "-c", "color.status=false"}, args...)
 	val, err := g.env.RunCommand(g.command, args...)
 	if err != nil {
 		return ""
@@ -695,7 +706,7 @@ func (g *Git) setGitHEADContext() {
 
 	getPrettyNameOrigin := func(file string) string {
 		var origin string
-		head := g.FileContents(g.workingDir, file)
+		head := g.FileContents(g.mainSCMDir, file)
 		if head == "detached HEAD" {
 			origin = formatDetached()
 		} else {
@@ -706,11 +717,11 @@ func (g *Git) setGitHEADContext() {
 	}
 
 	parseInt := func(file string) int {
-		val, _ := strconv.Atoi(g.FileContents(g.workingDir, file))
+		val, _ := strconv.Atoi(g.FileContents(g.mainSCMDir, file))
 		return val
 	}
 
-	if g.env.HasFolder(g.workingDir + "/rebase-merge") {
+	if g.env.HasFolder(g.mainSCMDir + "/rebase-merge") {
 		head := getPrettyNameOrigin("rebase-merge/head-name")
 		onto := g.getGitRefFileSymbolicName("rebase-merge/onto")
 		onto = g.formatBranch(onto)
@@ -729,7 +740,7 @@ func (g *Git) setGitHEADContext() {
 		return
 	}
 
-	if g.env.HasFolder(g.workingDir + "/rebase-apply") {
+	if g.env.HasFolder(g.mainSCMDir + "/rebase-apply") {
 		head := getPrettyNameOrigin("rebase-apply/head-name")
 		current := parseInt("rebase-apply/next")
 		total := parseInt("rebase-apply/last")
@@ -751,7 +762,7 @@ func (g *Git) setGitHEADContext() {
 	if g.hasGitFile("MERGE_MSG") {
 		g.Merge = true
 		icon := g.props.GetString(MergeIcon, "\uE727 ")
-		mergeContext := g.FileContents(g.workingDir, "MERGE_MSG")
+		mergeContext := g.FileContents(g.mainSCMDir, "MERGE_MSG")
 		matches := regex.FindNamedRegexMatch(`Merge (remote-tracking )?(?P<type>branch|commit|tag) '(?P<theirs>.*)'`, mergeContext)
 		// head := g.getGitRefFileSymbolicName("ORIG_HEAD")
 		if matches != nil && matches["theirs"] != "" {
@@ -779,7 +790,7 @@ func (g *Git) setGitHEADContext() {
 	// the todo file.
 	if g.hasGitFile("CHERRY_PICK_HEAD") {
 		g.CherryPick = true
-		sha := g.FileContents(g.workingDir, "CHERRY_PICK_HEAD")
+		sha := g.FileContents(g.mainSCMDir, "CHERRY_PICK_HEAD")
 		cherry := g.props.GetString(CherryPickIcon, "\uE29B ")
 		g.HEAD = fmt.Sprintf("%s%s%s onto %s", cherry, commitIcon, g.formatSHA(sha), formatDetached())
 		return
@@ -787,14 +798,14 @@ func (g *Git) setGitHEADContext() {
 
 	if g.hasGitFile("REVERT_HEAD") {
 		g.Revert = true
-		sha := g.FileContents(g.workingDir, "REVERT_HEAD")
+		sha := g.FileContents(g.mainSCMDir, "REVERT_HEAD")
 		revert := g.props.GetString(RevertIcon, "\uF0E2 ")
 		g.HEAD = fmt.Sprintf("%s%s%s onto %s", revert, commitIcon, g.formatSHA(sha), formatDetached())
 		return
 	}
 
 	if g.hasGitFile("sequencer/todo") {
-		todo := g.FileContents(g.workingDir, "sequencer/todo")
+		todo := g.FileContents(g.mainSCMDir, "sequencer/todo")
 		matches := regex.FindNamedRegexMatch(`^(?P<action>p|pick|revert)\s+(?P<sha>\S+)`, todo)
 		if matches != nil && matches["sha"] != "" {
 			action := matches["action"]
@@ -825,18 +836,18 @@ func (g *Git) formatSHA(sha string) string {
 }
 
 func (g *Git) hasGitFile(file string) bool {
-	return g.env.HasFilesInDir(g.workingDir, file)
+	return g.env.HasFilesInDir(g.mainSCMDir, file)
 }
 
 func (g *Git) getGitRefFileSymbolicName(refFile string) string {
-	ref := g.FileContents(g.workingDir, refFile)
+	ref := g.FileContents(g.mainSCMDir, refFile)
 	return g.getGitCommandOutput("name-rev", "--name-only", "--exclude=tags/*", ref)
 }
 
 func (g *Git) setPrettyHEADName() {
 	// we didn't fetch status, fallback to parsing the HEAD file
 	if len(g.ShortHash) == 0 {
-		HEADRef := g.FileContents(g.workingDir, "HEAD")
+		HEADRef := g.FileContents(g.mainSCMDir, "HEAD")
 		g.Detached = !strings.HasPrefix(HEADRef, "ref:")
 		if strings.HasPrefix(HEADRef, BRANCHPREFIX) {
 			branchName := strings.TrimPrefix(HEADRef, BRANCHPREFIX)
@@ -873,10 +884,10 @@ func (g *Git) WorktreeCount() int {
 	if g.worktreeCount > 0 {
 		return g.worktreeCount
 	}
-	if !g.env.HasFolder(g.rootDir + "/worktrees") {
+	if !g.env.HasFolder(g.scmDir + "/worktrees") {
 		return 0
 	}
-	worktreeFolders := g.env.LsDir(g.rootDir + "/worktrees")
+	worktreeFolders := g.env.LsDir(g.scmDir + "/worktrees")
 	var count int
 	for _, folder := range worktreeFolders {
 		if folder.IsDir() {
@@ -891,7 +902,7 @@ func (g *Git) getRemoteURL() string {
 	if len(upstream) == 0 {
 		upstream = "origin"
 	}
-	cfg, err := ini.Load(g.rootDir + "/config")
+	cfg, err := ini.Load(g.scmDir + "/config")
 	if err != nil {
 		return g.getGitCommandOutput("remote", "get-url", upstream)
 	}
@@ -905,7 +916,7 @@ func (g *Git) getRemoteURL() string {
 func (g *Git) Remotes() map[string]string {
 	var remotes = make(map[string]string)
 
-	location := filepath.Join(g.rootDir, "config")
+	location := filepath.Join(g.scmDir, "config")
 	config := g.env.FileContent(location)
 	cfg, err := ini.Load([]byte(config))
 	if err != nil {
@@ -941,7 +952,7 @@ func (g *Git) getSwitchMode(property properties.Property, gitSwitch, mode string
 		mode = val
 	}
 	// get the specific repo mode
-	if val := repoModes[g.realDir]; len(val) != 0 {
+	if val := repoModes[g.repoRootDir]; len(val) != 0 {
 		mode = val
 	}
 	if len(mode) == 0 {
@@ -952,12 +963,12 @@ func (g *Git) getSwitchMode(property properties.Property, gitSwitch, mode string
 
 func (g *Git) repoName() string {
 	if !g.IsWorkTree {
-		return path.Base(g.convertToLinuxPath(g.realDir))
+		return path.Base(g.convertToLinuxPath(g.repoRootDir))
 	}
 
-	ind := strings.LastIndex(g.workingDir, ".git/worktrees")
+	ind := strings.LastIndex(g.mainSCMDir, ".git/worktrees")
 	if ind > -1 {
-		return path.Base(g.workingDir[:ind])
+		return path.Base(g.mainSCMDir[:ind])
 	}
 
 	return ""

--- a/src/segments/git_test.go
+++ b/src/segments/git_test.go
@@ -59,7 +59,7 @@ func TestEnabledInWorkingDirectory(t *testing.T) {
 	g.Init(properties.Map{}, env)
 
 	assert.True(t, g.Enabled())
-	assert.Equal(t, fileInfo.Path, g.workingDir)
+	assert.Equal(t, fileInfo.Path, g.mainSCMDir)
 }
 
 func TestResolveEmptyGitPath(t *testing.T) {
@@ -139,9 +139,9 @@ func TestEnabledInWorktree(t *testing.T) {
 		g.Init(properties.Map{}, env)
 
 		assert.Equal(t, tc.ExpectedEnabled, g.hasWorktree(fileInfo), tc.Case)
-		assert.Equal(t, tc.ExpectedWorkingFolder, g.workingDir, tc.Case)
-		assert.Equal(t, tc.ExpectedRealFolder, g.realDir, tc.Case)
-		assert.Equal(t, tc.ExpectedRootFolder, g.rootDir, tc.Case)
+		assert.Equal(t, tc.ExpectedWorkingFolder, g.mainSCMDir, tc.Case)
+		assert.Equal(t, tc.ExpectedRealFolder, g.repoRootDir, tc.Case)
+		assert.Equal(t, tc.ExpectedRootFolder, g.scmDir, tc.Case)
 	}
 }
 
@@ -629,7 +629,7 @@ func TestGetStashContextZeroEntries(t *testing.T) {
 
 		g := &Git{
 			scm: scm{
-				workingDir: "",
+				mainSCMDir: "",
 			},
 		}
 		g.Init(properties.Map{}, env)
@@ -934,7 +934,7 @@ func TestGitUntrackedMode(t *testing.T) {
 
 		g := &Git{
 			scm: scm{
-				realDir: "foo",
+				repoRootDir: "foo",
 			},
 		}
 		g.Init(props, new(mock.Environment))
@@ -979,7 +979,7 @@ func TestGitIgnoreSubmodules(t *testing.T) {
 
 		g := &Git{
 			scm: scm{
-				realDir: "foo",
+				repoRootDir: "foo",
 			},
 		}
 		g.Init(props, new(mock.Environment))
@@ -1165,7 +1165,7 @@ func TestGitRemotes(t *testing.T) {
 
 		g := &Git{
 			scm: scm{
-				realDir: "foo",
+				repoRootDir: "foo",
 			},
 		}
 		g.Init(properties.Map{}, env)
@@ -1210,8 +1210,8 @@ func TestGitRepoName(t *testing.T) {
 
 		g := &Git{
 			scm: scm{
-				realDir:    tc.RealDir,
-				workingDir: tc.WorkingDir,
+				repoRootDir: tc.RealDir,
+				mainSCMDir:  tc.WorkingDir,
 			},
 			IsWorkTree: tc.IsWorkTree,
 		}

--- a/src/segments/mercurial.go
+++ b/src/segments/mercurial.go
@@ -83,10 +83,10 @@ func (hg *Mercurial) shouldDisplay() bool {
 
 	hg.setDir(hgdir.ParentFolder)
 
-	hg.workingDir = hgdir.Path
-	hg.rootDir = hgdir.Path
+	hg.mainSCMDir = hgdir.Path
+	hg.scmDir = hgdir.Path
 	// convert the worktree file path to a windows one when in a WSL shared folder
-	hg.realDir = strings.TrimSuffix(hg.convertToWindowsPath(hgdir.Path), "/.hg")
+	hg.repoRootDir = strings.TrimSuffix(hg.convertToWindowsPath(hgdir.Path), "/.hg")
 	return true
 }
 
@@ -165,7 +165,7 @@ func RemoveAtIndex(s []string, index int) []string {
 }
 
 func (hg *Mercurial) getHgCommandOutput(command string, args ...string) string {
-	args = append([]string{"-R", hg.realDir, command}, args...)
+	args = append([]string{"-R", hg.repoRootDir, command}, args...)
 	val, err := hg.env.RunCommand(hg.command, args...)
 	if err != nil {
 		return ""

--- a/src/segments/mercurial_test.go
+++ b/src/segments/mercurial_test.go
@@ -42,8 +42,8 @@ func TestMercurialEnabledInWorkingDirectory(t *testing.T) {
 	hg.Init(properties.Map{}, env)
 
 	assert.True(t, hg.Enabled())
-	assert.Equal(t, fileInfo.Path, hg.workingDir)
-	assert.Equal(t, fileInfo.Path, hg.realDir)
+	assert.Equal(t, fileInfo.Path, hg.mainSCMDir)
+	assert.Equal(t, fileInfo.Path, hg.repoRootDir)
 }
 
 func TestMercurialGetIdInfo(t *testing.T) {
@@ -159,8 +159,8 @@ A Added.File
 		}
 
 		assert.True(t, hg.Enabled())
-		assert.Equal(t, fileInfo.Path, hg.workingDir)
-		assert.Equal(t, fileInfo.Path, hg.realDir)
+		assert.Equal(t, fileInfo.Path, hg.mainSCMDir)
+		assert.Equal(t, fileInfo.Path, hg.repoRootDir)
 		assert.Equal(t, tc.ExpectedWorking, hg.Working, tc.Case)
 		assert.Equal(t, tc.ExpectedBranch, hg.Branch, tc.Case)
 		assert.Equal(t, tc.ExpectedChangeSetID, hg.ChangeSetID, tc.Case)

--- a/src/segments/sapling.go
+++ b/src/segments/sapling.go
@@ -82,11 +82,11 @@ func (sl *Sapling) shouldDisplay() bool {
 		return false
 	}
 
-	sl.workingDir = slDir.Path
-	sl.rootDir = slDir.Path
+	sl.mainSCMDir = slDir.Path
+	sl.scmDir = slDir.Path
 	// convert the worktree file path to a windows one when in a WSL shared folder
-	sl.realDir = strings.TrimSuffix(sl.convertToWindowsPath(slDir.Path), "/.sl")
-	sl.RepoName = path.Base(sl.convertToLinuxPath(sl.realDir))
+	sl.repoRootDir = strings.TrimSuffix(sl.convertToWindowsPath(slDir.Path), "/.sl")
+	sl.RepoName = path.Base(sl.convertToLinuxPath(sl.repoRootDir))
 	sl.setDir(slDir.Path)
 
 	return true

--- a/src/segments/sapling_test.go
+++ b/src/segments/sapling_test.go
@@ -151,9 +151,9 @@ func TestShouldDisplay(t *testing.T) {
 		got := sl.shouldDisplay()
 		assert.Equal(t, tc.Expected, got, tc.Case)
 		if tc.Expected {
-			assert.Equal(t, "/sapling/repo/.sl", sl.workingDir, tc.Case)
-			assert.Equal(t, "/sapling/repo/.sl", sl.rootDir, tc.Case)
-			assert.Equal(t, "/sapling/repo", sl.realDir, tc.Case)
+			assert.Equal(t, "/sapling/repo/.sl", sl.mainSCMDir, tc.Case)
+			assert.Equal(t, "/sapling/repo/.sl", sl.scmDir, tc.Case)
+			assert.Equal(t, "/sapling/repo", sl.repoRootDir, tc.Case)
 			assert.Equal(t, "repo", sl.RepoName, tc.Case)
 		}
 	}

--- a/src/segments/scm.go
+++ b/src/segments/scm.go
@@ -84,9 +84,9 @@ type scm struct {
 
 	Dir             string
 	RepoName        string
-	workingDir      string
-	rootDir         string
-	realDir         string
+	mainSCMDir      string
+	scmDir          string
+	repoRootDir     string
 	command         string
 	IsWslSharedPath bool
 	CommandMissing  bool

--- a/src/segments/svn.go
+++ b/src/segments/svn.go
@@ -79,21 +79,21 @@ func (s *Svn) shouldDisplay() bool {
 	}
 
 	if Svndir.IsDir {
-		s.workingDir = Svndir.Path
-		s.rootDir = Svndir.Path
+		s.mainSCMDir = Svndir.Path
+		s.scmDir = Svndir.Path
 		// convert the worktree file path to a windows one when in a WSL shared folder
-		s.realDir = strings.TrimSuffix(s.convertToWindowsPath(Svndir.Path), "/.svn")
+		s.repoRootDir = strings.TrimSuffix(s.convertToWindowsPath(Svndir.Path), "/.svn")
 		return true
 	}
 
 	// handle worktree
-	s.rootDir = Svndir.Path
+	s.scmDir = Svndir.Path
 	dirPointer := strings.Trim(s.env.FileContent(Svndir.Path), " \r\n")
 	matches := regex.FindNamedRegexMatch(`^Svndir: (?P<dir>.*)$`, dirPointer)
 	if matches != nil && matches["dir"] != "" {
 		// if we open a worktree file in a WSL shared folder, we have to convert it back
 		// to the mounted path
-		s.workingDir = s.convertToLinuxPath(matches["dir"])
+		s.mainSCMDir = s.convertToLinuxPath(matches["dir"])
 	}
 	return false
 }
@@ -141,7 +141,7 @@ func (s *Svn) Repo() string {
 }
 
 func (s *Svn) getSvnCommandOutput(command string, args ...string) string {
-	args = append([]string{command, s.realDir}, args...)
+	args = append([]string{command, s.repoRootDir}, args...)
 	val, err := s.env.RunCommand(s.command, args...)
 	if err != nil {
 		return ""

--- a/src/segments/svn_test.go
+++ b/src/segments/svn_test.go
@@ -44,8 +44,8 @@ func TestSvnEnabledInWorkingDirectory(t *testing.T) {
 	s.Init(properties.Map{}, env)
 
 	assert.True(t, s.Enabled())
-	assert.Equal(t, fileInfo.Path, s.workingDir)
-	assert.Equal(t, fileInfo.Path, s.realDir)
+	assert.Equal(t, fileInfo.Path, s.mainSCMDir)
+	assert.Equal(t, fileInfo.Path, s.repoRootDir)
 }
 
 func TestSvnTemplateString(t *testing.T) {

